### PR TITLE
Allow Setting for Host DNS Entries

### PR DESCRIPTION
**Pull request by eparis**
This will allow one to set host based DNS entries. So I can say
host=host1, ip=1.1.1.1
host=host1, ip=1.1.1.2
host=host2, ip=1.1.1.1
And It would work the way you expect...

### CODE REVIEW

**Comment by eparis** [comment]
this also includes the DNS rework from #376

File: libvirt/resource_libvirt_network.go [hunk 1]
```diff
@@ -105,6 +148,65 @@ func resourceLibvirtNetwork() *schema.Resource {
...
+//
+// resource "libvirt_network" "k8snet" {
+//   ...
+//   dns = [{
+//     hosts = [ "${flatten(data.libvirt_network_dns_host_template.k8smasters.*.rendered)}" ]
+//   }]
+//   ...
+// }
+//
+func datasourceLibvirtNetworkDNSHostTemplate() *schema.Resource {
```
Comment (MalloZup) [inline libvirt/resource_libvirt_network.go]: i am thinking that if we introduce a datasource resource, would make sense to me to create a new file 
as example:
https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/data_source_aws_acmpca_certificate_authority_test.go

https://github.com/terraform-providers/terraform-provider-google/tree/master/google

**Comment by eparis** [comment]
@MalloZup hopefully I learned enough from #376 that this review is easier.

**Comment by MalloZup** [comment]
@eparis i like your pr with the data_source especially in this way we could also start a new pattern for refactoring other resources ( i plan to investiage more on this with cloud_init user_data).

to me seems good, i will just wait that @dmacvicar approve the #376  so this pr should be easier to review (showing only new-diff). So far looks good :+1:

**Comment by MalloZup** [comment]
@eparis can you rebase with upstream so we see on the diff only the real changes TIA :japanese_castle: :tulip:

**Comment by eparis** [comment]
rebased.  thank you!

**Comment by MalloZup** [comment]
@eparis ok thx, i will fix the small thing is failing  :D i have renamed the function you have introduced with the last pr it so that is why the test are failing :smile:

**Comment by MalloZup** [comment]
@eparis  i will review it more in deep tomorrow with a fresh mind. sofar looks nice to me but i want to try it out with time. thank you for your patience :+1:

**Comment by eparis** [comment]
@MalloZup Patience? You've been one of the most responsive and helpful maintainers I've ever dealt with! Thank you for helping me get it right.

**Comment by MalloZup** [comment]
merging. thx @eparis  for the contribution! i hope it will be helpful for your projects and c u next contrib :+1: :smile:

**Pull request merged**